torch_xla/csrc/init_python_bindings.cpp

```diff
@@ -1501,9 +1501,12 @@ void InitXlaModuleBindings(py::module m) {
       // If the at::Tensor data is not present, we need to re-download the
       // tensor from the physical device to CPU. In that case, the value
       // must be present on the backend device.
-      XLA_CHECK(xtensor->GetXlaData() != nullptr &&
-                xtensor->CurrentDataHandle()->HasValue())
-          << "Cannot shard tensor. Data not present on any device.";
+      {
+        XLAGraphExecutor::Get()->DeviceBarrier(xtensor->GetDevice());
```

Comment:
why do we need to add a barrier here?

Reply:
Let's have a offline discussion, I would like to understand as well

Reply:
If we don't use the barrier, GetXlaData will error out if there is ongoing operation on the device.